builder/dockerfile/command/commands.go

```diff
@@ -0,0 +1,232 @@
+}
+
+type EnvCommand struct {
+	OriginalSource string
+	Env            KeyValuePairs // kvp slice instead of map to preserve ordering
+}
+
+func (c *EnvCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type MaintainerCommand struct {
+	OriginalSource string
+	Maintainer     string
+}
+
+func (c *MaintainerCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type LabelCommand struct {
+	OriginalSource string
+	Labels         KeyValuePairs // kvp slice instead of map to preserve ordering
+}
+
+func (c *LabelCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type AddCommand struct {
+	OriginalSource string
+	Srcs           []string
+	Dest           string
+}
+
+func (c *AddCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type CopyCommand struct {
+	OriginalSource string
+	Srcs           []string
+	Dest           string
+	From           string
+}
+
+func (c *CopyCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type FromCommand struct {
+	OriginalSource string
+	BaseName       string
+	StageName      string
+}
+
+func (c *FromCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type OnbuildCommand struct {
+	OriginalSource string
+	Expression     string
+}
+
+func (c *OnbuildCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type WorkdirCommand struct {
+	OriginalSource string
+	Path           string
+}
+
+func (c *WorkdirCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type RunCommand struct {
+	OriginalSource string
+	Expression     strslice.StrSlice
+	PrependShell   bool
+}
+
+func (c *RunCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type CmdCommand struct {
+	OriginalSource string
+	Cmd            strslice.StrSlice
+	PrependShell   bool
+}
+
+func (c *CmdCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type HealthCheckCommand struct {
+	OriginalSource string
+	Health         *container.HealthConfig
+}
+
+func (c *HealthCheckCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type EntrypointCommand struct {
+	OriginalSource string
+	CmdLine        strslice.StrSlice
+	Discard        bool
+	PrependShell   bool
+}
+
+func (c *EntrypointCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type ExposeCommand struct {
+	OriginalSource string
+	Ports          []string
+}
+
+func (c *ExposeCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type UserCommand struct {
+	OriginalSource string
+	User           string
+}
+
+func (c *UserCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type VolumeCommand struct {
+	OriginalSource string
+	Volumes        []string
+}
+
+func (c *VolumeCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type StopSignalCommand struct {
+	OriginalSource string
+	Sig            string
+}
+
+func (c *StopSignalCommand) SourceCode() string {
+	return c.OriginalSource
+}
+
+type ArgCommand struct {
+	OriginalSource string
+	Arg            string
```

Comment:
The ArgCommand is just there to add a commit point in the build in cases where `arg` is after the first `from`. Its `Arg` field value is `key=value` if it is a key with a default value. But I agree it is a bit confusing on review. I'll rename the field to make it more explicit.